sockeye/inference.py

```diff
@@ -92,11 +92,23 @@ def initialize(self, max_input_length: int, get_max_output_length_function: Call
         :param get_max_output_length_function: Callable to compute maximum output length.
         """
         self.max_input_length = max_input_length
-        if self.max_input_length != self.config.max_seq_len_source:
+        if self.max_input_length > self.training_max_seq_len_source:
             logger.warning("Model was trained with max_seq_len_source=%d, but using max_input_len=%d.",
-                           self.config.max_seq_len_source, self.max_input_length)
+                           self.training_max_seq_len_source, self.max_input_length)
         self.get_max_output_length = get_max_output_length_function
 
+        # check the maximum supported length of the encoder & decoder:
+        if self.max_supported_seq_len_source is not None:
+            utils.check_condition(self.max_input_length <= self.max_supported_seq_len_source,
+                                  "Encoder only supports a maximum length of %d" % self.max_supported_seq_len_source)
+        decoder_max_len = self.get_max_output_length(max_input_length)
```

Comment:
this can be moved inside the if statement below right?

Reply:
done